### PR TITLE
Improve config code

### PR DESCRIPTION
**Pull request by caarlos0**
- tried to make the code more idiomatic and split it in more functions.
- added more tests and remove an unused function

### CODE REVIEW

**Comment by caarlos0** [comment]
cc/ @jspc

File: config/config_test.go [hunk 1]
```diff
@@ -18,57 +19,88 @@ func TestFixConfig(t *testing.T) {
...
 	assert.Equal([]string{}, config.Files)
 }
 
-func TestFixConfigUSENMarkdown(t *testing.T) {
+func TestFillFilesUSENMarkdown(t *testing.T) {
 	assert := assert.New(t)
 
 	cwd, _ := os.Getwd()
 	os.Chdir("./.test/1")
+	defer os.Chdir(cwd)
```
Comment (jspc) [inline config/config_test.go]: 👍

File: config/config_test.go [hunk 2]
```diff
@@ -18,57 +19,88 @@ func TestFixConfig(t *testing.T) {
...
+	config := ProjectConfig{}
+	err := config.fillFiles()
 
-	os.Chdir(cwd)
+	assert.NoError(err)
+	assert.Equal([]string{"LICENSE.md", "README.md"}, config.Files)
 }
 
-func TestFixConfigRealENMarkdown(t *testing.T) {
+func TestFillFilesRealENMarkdown(t *testing.T) {
```
Comment (jspc) [inline config/config_test.go]: Good to see Real English snuck in

File: config/config.go [hunk 3]
```diff
@@ -56,33 +55,43 @@ func Load(file string) (config ProjectConfig, err error) {
...
-	config, err = fillGitData(config)
-	if err != nil {
+	config.fillBasicData()
+	if err := config.fillFiles(); err != nil {
+		return config, err
+	}
+	if err := config.fillGitData(); err != nil {
 		return config, err
 	}
+	return config, config.validade()
```
Comment (jspc) [inline config/config.go]: I suspect goreportcard will complain about validade over validate.

File: config/config.go [hunk 4]
```diff
@@ -56,33 +55,43 @@ func Load(file string) (config ProjectConfig, err error) {
...
+}
+
+func (config *ProjectConfig) validade() (err error) {
 	if config.BinaryName == "" {
-		return config, errors.New("missing binary_name")
+		return errors.New("missing binary_name")
```
Comment (jspc) [inline config/config.go]: You could, if you wanted, replace these `errors.New` statements with `fmt.Errorf`.

But that seems way out of scope for this PR

**Pull request merged**